ui/src/app/views/account-creation-modals/component.tsx

```diff
@@ -0,0 +1,231 @@
+  }
+
+  render() {
+    return <React.Fragment>
+      {this.props.resend &&
+      <Modal id='resend-instructions'>
+        <ModalTitle>Resend Instructions</ModalTitle>
+        <ModalFooter>
+          <button type='button' className='btn btn-outline'
+                  onClick={() => this.close()}>Cancel</button>
+          <button type='button' id='resend_instructions'
+                  className='btn btn-primary'
+                  onClick={() => this.send()}>Send</button>
+        </ModalFooter>
+      </Modal>}
+    </React.Fragment>;
+  }
+
+}
+
+class AccountCreationUpdateModalReact extends React.Component<any, any> {
+
+  state: {
+    contactEmail: string,
+    emailOffFocus: boolean,
+  };
+  props: {
+    username: string,
+    creationNonce: string,
+    passNewEmail: Function,
+    update: boolean,
+    closeFunction: Function,
+  };
+
+  constructor(props: Object) {
+    super(props);
+    this.state = {
+      contactEmail: '',
+      emailOffFocus: true,
+    };
+  }
+
+  updateAndSend(): void {
+    if (this.contactEmailInvalidError()) {
+      return;
+    }
+    this.props.passNewEmail(this.state.contactEmail);
+    const request: UpdateContactEmailRequest = {
+      username: this.props.username,
+      contactEmail: this.state.contactEmail,
+      creationNonce: this.props.creationNonce
+    };
+    const args: FetchArgs = ProfileApiFetchParamCreator().updateContactEmail(request);
+    fetch(fullUrl(args.url), args.options)
+      .then(handleErrors)
+      .catch(error => console.log(error));
+    this.props.closeFunction();
+  }
+
+  leaveFocusEmail(): void {
+    this.setState({emailOffFocus: true});
+  }
+
+  enterFocusEmail(): void { this.setState({emailOffFocus: false}); }
+
+  updateContactEmail(evt: React.ChangeEvent<HTMLInputElement>): void {
+    this.setState({contactEmail: evt.target.value});
+  }
+
+  contactEmailInvalidError(): boolean {
+    return !(new RegExp(/^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/).test(this.state.contactEmail));
+  }
+
+  showEmailValidationError(): boolean {
+    if (isBlank(this.state.contactEmail) || !this.state.emailOffFocus) {
+      return false;
+    }
+    return this.contactEmailInvalidError();
+  }
+
+  close(): void {
+    this.props.closeFunction();
+  }
+
+  render() {
+    return <React.Fragment>
+      {this.props.update &&
+      <Modal>
+        <ModalTitle>Change contact email</ModalTitle>
+        <ModalBody>
+          <table style={{width: '100%'}}>
+            <tbody>
+            <tr>
+              <td><label>Contact Email:</label></td>
+              <td style={{width: '70%'}}><FieldInput
+                id='change-contact-email'
+                style={this.showEmailValidationError() ? styles.unsuccessfulInput : {}}
+                onChange={(e) => this.setState({contactEmail: e.target.value})}
+                onBlur={() => this.leaveFocusEmail()}
+                onFocus={() => this.enterFocusEmail()}
+              >
+              </FieldInput>
```

Comment:
FYI, if an element is empty, you can use a self-closing tag syntax (works for any element, even custom ones)